File: ruby/redis/redis_connect.rb
require 'redis' #using the redis-rb, A Ruby client library for Redis
require 'connection_pool' //Generic connection pooling for Ruby


#If no object is available in the pool within :timeout seconds, it  will raise an error
Redis.current = ConnectionPool.new(size: 10, timeout: 5) do
 	#All timeout values are specified in seconds
	redis = Redis.new(
		:host => "pub-redis-13628.us-east-1-2.1.ec2.garantiadata.com", 
		:port => 13628, 
		:db => 0, 
		:password => "rs87362c",
		:connect_timeout => 1.0,
		:read_timeout => 1.0,
		:write_timeout   => 1.0
	)
	redis.set('foo', 'bar');
	value = redis.get('foo');
	puts  value
end

File: python/redis/redis-connect.py
import redis

print("\nconnecting to redis labs cloud...")	

pool = redis.ConnectionPool(host='pub-redis-11458.us-east-mz.1.ec2.garantiadata.com', 
	port=11458, password='rs87362c',
	max_connections=1)
r = redis.Redis(connection_pool=pool)

''' without connection pooling:
r = redis.StrictRedis(host='pub-redis-13628.us-east-1-2.1.ec2.garantiadata.com', port=13628,password='rs87362c')
'''

'''with SSL:
r = redis.Redis( url='rediss://:password@hostname:port/0',
    password='password',
    ssl_keyfile='path_to_keyfile/garantia_user_private.key',
    ssl_certfile='path_to_certfile/garantia_user.crt',
    ssl_cert_reqs='required',
    ssl_ca_certs='path_to_ca_certfile/garantia_ca.pem')
'''

print("\nset test key...")
print("\n>r.set(`foo`, `bar`)")
r.set('eran', 'bar')

print("\nget test key...")
print("\n>r.get(`foo`)")
print(">foo=" + r.get('foo'))

print("\ndone.")
